proto-lens-protoc/src/Data/ProtoLens/Compiler/Generate.hs

```diff
@@ -259,80 +259,193 @@ generateEnumExports syntaxType e = [exportAll n, exportWith n aliases]
 
 generateEnumDecls :: SyntaxType -> EnumInfo Name -> [Decl]
 generateEnumDecls Proto3 info =
-    -- newtype Foo = Foo Int32
-    [ newtypeDecl dataName "Data.Int.Int32"
-        $ deriving' ["Prelude.Eq", "Prelude.Ord", "Prelude.Enum", "Prelude.Bounded"]
-
-    -- instance Show Foo where
-    --    showsPrec _ Value0 = "Value0" -- the Haskell name
-    --    showsPrec p (Foo k) = showParen (p > 10)
-    --                            $ showString "toEnum " . shows k
-    , instDecl [] ("Prelude.Show" `ihApp` [dataType])
-        [ [ match "showsPrec" [pWildCard, pApp (unQual n) []]
-               $ "Prelude.showString" @@ stringExp (prettyPrint n)
-          | n <- map enumValueName $ enumValues info
+    -- data FooEnum
+    --     = Enum1
+    --     | Enum2
+    --     | FooEnum'Unrecognized !FooEnum'UnrecognizedValue
+    --   deriving (Prelude.Show, Prelude.Eq, Prelude.Ord, Prelude.Read)
+    [ dataDecl dataName
+        (  (flip conDecl [] <$> constructorNames)
+        ++ [conDecl unrecognizedName [tyCon $ unQual unrecognizedValueName]]
+        )
+        $ deriving' ["Prelude.Show", "Prelude.Eq", "Prelude.Ord, Prelude.Read"]
+
+    -- newtype FooEnum'UnrecognizedValue = FooEnum'UnrecognizedValue Data.Int.Int32
+    --   deriving (Prelude.Eq, Prelude.Ord, Prelude.Show, Prelude.Read)
+    , newtypeDecl unrecognizedValueName
+       "Data.Int.Int32"
+        $ deriving' ["Prelude.Eq", "Prelude.Ord", "Prelude.Show, Prelude.Read"]
```

Comment:
Can we remove the `Read` instance, to make sure it's not being used accidentally?

(Also because we'd like `toEnum/maybeToEnum` to be the only way to construct these values)

Up to you.